src/proj.h

```diff
@@ -559,27 +568,490 @@ PJ_OBJ_LIST PROJ_DLL *proj_obj_create_from_name(PJ_CONTEXT *ctx,
+                                            PJ_OPERATION_FACTORY_CONTEXT *ctxt,
+                                            double accuracy);
+
+void PROJ_DLL proj_operation_factory_context_set_area_of_interest(
+                                            PJ_OPERATION_FACTORY_CONTEXT *ctxt,
+                                            double west_lon_degree,
+                                            double south_lat_degree,
+                                            double east_lon_degree,
+                                            double north_lat_degree);
+
+/** Specify how source and target CRS extent should be used to restrict
+  * candidate operations (only taken into account if no explicit area of
+  * interest is specified. */
+typedef enum
+{
+    /** Ignore CRS extent */
+    PJ_CRS_EXTENT_NONE,
+
+    /** Test coordinate operation extent against both CRS extent. */
+    PJ_CRS_EXTENT_BOTH,
+
+    /** Test coordinate operation extent against the intersection of both
+        CRS extent. */
+    PJ_CRS_EXTENT_INTERSECTION,
+
+    /** Test coordinate operation against the smallest of both CRS extent. */
+    PJ_CRS_EXTENT_SMALLEST
+} PROJ_CRS_EXTENT_USE;
+
+void PROJ_DLL proj_operation_factory_context_set_crs_extent_use(
+                                            PJ_OPERATION_FACTORY_CONTEXT *ctxt,
+                                            PROJ_CRS_EXTENT_USE use);
+
+/** Spatial criterion to restrict candiate operations. */
+typedef enum {
+    /** The area of validity of transforms should strictly contain the
+        * are of interest. */
+    PROJ_SPATIAL_CRITERION_STRICT_CONTAINMENT,
+
+    /** The area of validity of transforms should at least intersect the
+        * area of interest. */
+    PROJ_SPATIAL_CRITERION_PARTIAL_INTERSECTION
+} PROJ_SPATIAL_CRITERION;
+
+void PROJ_DLL proj_operation_factory_context_set_spatial_criterion(
+                                            PJ_OPERATION_FACTORY_CONTEXT *ctxt,
+                                            PROJ_SPATIAL_CRITERION criterion);
+
+
+/** Describe how grid availability is used. */
+typedef enum {
+    /** Grid availability is only used for sorting results. Operations
+        * where some grids are missing will be sorted last. */
+    PROJ_GRID_AVAILABILITY_USED_FOR_SORTING,
+
+    /** Completely discard an operation if a required grid is missing. */
+    PROJ_GRID_AVAILABILITY_DISCARD_OPERATION_IF_MISSING_GRID,
+
+    /** Ignore grid availability at all. Results will be presented as if
+        * all grids were available. */
+    PROJ_GRID_AVAILABILITY_IGNORED,
+} PROJ_GRID_AVAILABILITY_USE;
+
+void PROJ_DLL proj_operation_factory_context_set_grid_availability_use(
+                                            PJ_OPERATION_FACTORY_CONTEXT *ctxt,
+                                            PROJ_GRID_AVAILABILITY_USE use);
+
+void PROJ_DLL proj_operation_factory_context_set_use_proj_alternative_grid_names(
+    PJ_OPERATION_FACTORY_CONTEXT *ctxt,
```

Comment:
Usually we just use `ctx` for the argument name of contexts. Is the extra "t" in `ctxt` signaling that this context is different from a `PJ_CONTEXT`?